where-to-travel/src/main/webapp/script.js

```diff
@@ -449,76 +452,90 @@ function clearPlaces() {
-          }
+ function getPlacesFromTime(time) {
+  return new Promise(async function(resolve, reject){    
+    const localNonce = globalNonce = new Object(); 
+    // First try one bounding box around user's location
+    let place_candidates = await getPlacesFromDirection(home.lat, home.lng);
+    let filterResults = await filterByTime(time, place_candidates);
+
+    // If most recent request has changed, don't continue with search
+    if (localNonce != globalNonce) {
+      reject('Cancelled');
+    }
 
-          if (lngSpread != 0) {
-            lngSpread = lngSpread < 0 ? lngSpread + 1 : lngSpread - 1;
-          }
-        } else {
-          if (latSpread != 0) {
-            latSpread = latSpread < 0 ? latSpread - 1 : latSpread + 1;
-          }
+    let places = filterResults.places;
+
+    // Initial distance from the user's location for the bounding boxes
+    const initSpread = Math.max(1, Math.ceil(time/7200));
+    let attempts = 0;
+
+    /* Each direction is represented by a pair with the first element added
+       to the user's lat and the second element added to the user's lng */
+    let directions = [
+      [initSpread,0], //North
+      [0,initSpread], // East
+      [0,-initSpread], // West
+      [-initSpread,0], // South
+      [initSpread, -initSpread], // Northwest
+      [initSpread, initSpread], // Northeast
+      [-initSpread, initSpread], // Southeast
+      [-initSpread, -initSpread] // Southwest
+    ];
+
+
+    while (attempts < ATTEMPTS_THRESHOLD && places.length < PLACES_THRESHOLD) {
+      let new_directions = [];
+
+      for (direction of directions) {
+        let latSpread = direction[0];
+        let lngSpread = direction[1];
+
+        let place_candidates = await getPlacesFromDirection(home.lat + latSpread, home.lng + lngSpread);
+        let filterResults = await filterByTime(time, place_candidates);
+        
+        // If most recent request has changed, don't continue with search
+        if (localNonce != globalNonce) {
+          reject('Cancelled');
+        }
+
+        places = places.concat(filterResults.places);
+
+        // If bounding box does not contain enough results, update position of box for next iteration
+        if (filterResults.places.length < DIRECTION_THRESHOLD) {
+          /* If average time in bounding box is greater than requested time, move bounding box closer
+             to user otherwise move bounding box farther away from user. */
+          if (filterResults.avg_time > time) {
+            if (latSpread != 0) {
+              latSpread = latSpread < 0 ? latSpread + 1 : latSpread - 1;
+            }
 
-          if (lngSpread != 0) {
-            lngSpread = lngSpread < 0 ? lngSpread - 1 : lngSpread + 1;
+            if (lngSpread != 0) {
+              lngSpread = lngSpread < 0 ? lngSpread + 1 : lngSpread - 1;
+            }
+          } else {
+            if (latSpread != 0) {
+              latSpread = latSpread < 0 ? latSpread - 1 : latSpread + 1;
```

Comment:
Could you make this an else if?

Reply:
I want both of the if statements at 517 and 521 to fall under the case that the avg_time from the filter results is greater than or equal to the requested time. It's possible that they might both be entered under the else branch so I think it would be easier to keep it as is. Let me know what you think!

Reply:
sounds good!